src/algorithms/digi/SiliconTrackerDigi.cc

```diff
@@ -90,6 +90,16 @@ void SiliconTrackerDigi::process(
 
     for (auto item : cell_hit_map) {
         raw_hits->push_back(item.second);
+
+        // set association
+        auto hitassoc = associations->create();
+        hitassoc.setWeight(1.0);
+        hitassoc.setRawHit(item.second);
+
+        for (const auto& sim_hit : *sim_hits) {
+                if(item.first == sim_hit.getCellID()) hitassoc.addToSimHits(sim_hit);
```

Comment:
I believe, having sim hits as a OneToMany relation [1] was a mistake. The idea behind having a weight is to specify level of contribution of a specific sim hit. If multiple sim hits are associated, user should expect multiple association records in the collection. I move to change this code to include one and only one sim hit per association, and eventually modify the EDM to conform to the standard.

cc @wdconinc @c-dilks

[1] https://github.com/eic/EDM4eic/blob/cc887f9b8ca78909ca88964ac661e1cafce51c62/edm4eic.yaml#L549

```suggestion
        for (const auto& sim_hit : *sim_hits) {
          // set association
          auto hitassoc = associations->create();
          hitassoc.setWeight(1.0);
          hitassoc.setRawHit(item.second);
          if(item.first == sim_hit.getCellID()) hitassoc.addToSimHits(sim_hit);
```